lib/jekyll/document.rb

```diff
@@ -483,6 +485,9 @@ def populate_title
         slug, ext = Regexp.last_match.captures
       end
 
+      # slugs shouldn't end with a period
+      slug.gsub!(%r!\.*\z!, "")
```

Comment:
You might be able to use `.chomp(".")` here instead. [String#chomp](https://ruby-doc.org/core-2.2.0/String.html#method-i-chomp)

Reply:
I intentionally did not use `String#chomp` here because `#chomp` removes only one trailing character.
```ruby
"hello-world...".chomp(".")           # => "hello-world.."
"hello-world...".gsub(%r!\.*\z!, "")  # => "hello-world"
```